src/target/jtag_scan.h

```diff
@@ -43,10 +43,10 @@ typedef struct jtag_dev_s {
 } jtag_dev_t;
 
 extern struct jtag_dev_s jtag_devs[JTAG_MAX_DEVS + 1];
-extern uint32_t jtag_dev_count;
+extern size_t jtag_dev_count;
```

Comment:
We'd made this a uint32_t because of the remote.c issues bought up by size_t, otherwise we'd have done the same. Likewise for jtag_add_device()

Reply:
Ah, my bad, i went a little ham with that

Reply:
We failed to document why we'd made the type choice we had, so there was nothing to indicate why it was a bad idea.. to be fair